lib/builderator/util.rb

```diff
@@ -60,8 +60,15 @@ def filter!(resources, filters = {})
       ##
       # AWS Clients
       ##
-      def ec2(region = Config.aws.region)
-        clients["ec2-#{region}"] ||= Aws::EC2::Client.new(:region => region)
+      def ec2(region = Config.aws.region, credentials=nil)
+        options = { :region => region }
+
+        # Don't memoize if supplying explicit credentials as it could be an assumed role for a remote account
+        if credentials.nil?
```

Comment:
Are you asking why the existing code doesn't work for me?  If so - first we build the AMI with a 'regular' EC2 client in us-east-1.  To tag it, we have to assume the IAM role through STS - this means a different EC2 client with different credentials, but still in us-east-1.

If you're asking why haven't I memoize'd the new EC2 client (with the assumed role), I guess we wouldn't be likely to reuse it anyway.